### ISSUE TITLE
Add PAG support to SD1.5

### ISSUE THREAD

**Issue by yiyixuxu** [huggingface/diffusers#8710]
We recently integrated PAG into diffusers! See this PR [here] (https://github.com/huggingface/diffusers/pull/7944) we added PAG to SDXL

we also want to add PAG support to SD1.5 pipelines! we will need:

- [x] StableDiffusionPAGPipeline (assigned to @shauray8, PR https://github.com/huggingface/diffusers/pull/8725)
- [ ] StableDiffusionPAGImg2ImgPipeline https://github.com/huggingface/diffusers/pull/9463
- [ ] StableDiffusionPAGInpaintPipeline
- [ ] StableDiffusionControlNetPAGInpaintPipeline (https://github.com/huggingface/diffusers/pull/8875)
- [x] StableDiffusionControlNetPAGPipeline (assigned to @tuanh123789 )
- [ ] StableDiffusionControlNetPAGImg2ImgPipeline (assigned to @Bhavay-2001 https://github.com/huggingface/diffusers/pull/8864)

1. You should put it under the [pag folder](https://github.com/huggingface/diffusers/tree/main/src/diffusers/pipelines/pag)
2. you can use the implementation of SDXL PAG pipelines as a reference (see this PRhttps://github.com/huggingface/diffusers/pull/7944 and you can find all the sdxl pag pipelines here https://github.com/huggingface/diffusers/tree/main/src/diffusers/pipelines/pag)
3. you need to add AutoPipeline so that you can use this API to create it
    ```python
       AutoPipelineForImage2Image.from_pretrained(repo_id, controlnet=controlnet, enable_pag=True ...)
   ```
4. tests and docs 

If you are interested in working on this, Let me know which pipeline(s) you want to work on:)

**Comment by yiyixuxu** [comment]
@blaisedmello sure!

**Comment by shauray8** [comment]
Looks rather interesting, adding support for StableDiffusionPAGPipeline.

**Comment by sajadn** [comment]
Hey, I'll work on `StableDiffusionPAGImg2ImgPipeline`

**Comment by tuanh123789** [comment]
Hi, can i take `StableDiffusionControlNetPAGPipeline`

**Comment by yiyixuxu** [comment]
@tuanh123789 sure!

**Comment by Bhavay-2001** [comment]
Hi @yiyixuxu, I think I can work on `StableDiffusionControlNetPAGImg2ImgPipeline`. Pls assign this to me.

**Comment by yiyixuxu** [comment]
@Bhavay-2001 sure!

**Comment by AbhinavJangra29** [comment]
@yiyixuxu  can i try ?

**Comment by darshil0805** [comment]
Hi, I'd like to work on StableDiffusionPAGInpaintPipeline if it's still open

**Comment by yiyixuxu** [comment]
@darshil0805 sure!

**Comment by darshil0805** [comment]
Hi @yiyixuxu , @a-r-r-o-w. I have opened a PR for SD Inpainting PAG Pipeline, can you review and help me to test this pipeline?

**Comment by crapthings** [comment]
how do i know the layer names?
only
up mid down?

**Comment by yiyixuxu** [comment]
cc @a-r-r-o-w, maybe we should update doc? (did we?)

**Comment by a-r-r-o-w** [comment]
> how do i know the layer names?
only
up mid down?

The correct usage is to look at either the modeling code, or state dict keys, to see what the layer naming looks like. Then you can specify either the full layer names or partial layer names as valid regular expressions when initializing the pipeline (via `pag_applied_layers`). For more advanced usage, you can take a look at helper methods in the PAGMixin.

See the docs and examples usages [here](https://huggingface.co/docs/diffusers/v0.30.3/en/api/pipelines/pag#diffusers.HunyuanDiTPAGPipeline).

TLDR; Our naming convention follows the pattern of `something_blocks.whatever` or `something_blocks.[layer_index}` or `blocks.{layer_index}` or `transformer_blocks.{layer_index}`, so the easiest way to enable them without looking into the state dict or modeling code is to specify something like `pag_applied_layers="blocks.(10|20|42)"` or `pag_applied_layers="mid"`

> cc @a-r-r-o-w, maybe we should update doc? (did we?)

I think we did, but I noticed that the introduction [docs](https://huggingface.co/docs/diffusers/v0.30.3/en/using-diffusers/pag?tasks=Text-to-image#general-tasks) are out of sync with the current usage, and I'm not sure if the current example works. I'll test and open a PR soon

**Comment by airookie17** [comment]
HI @yiyixuxu I'd like to work on `StableDiffusionPAGImg2ImgPipeline` if it's still open. Thanks

**Comment by yiyixuxu** [comment]
@airookie17 
I think I'm about to merge in this https://github.com/huggingface/diffusers/pull/9463
